src/test/kotlin/net/atos/zac/zoeken/IndexingServiceTest.kt

```diff
@@ -157,16 +163,46 @@ class IndexingServiceTest : BehaviorSpec({
             }
         }
 
-        When("error occurs during indexing") {
+        When("adding beans in Solr errors") {
             val solrException = SolrServerException("Solr exception")
             every { solrClient.addBeans(any<Collection<*>>()) } throws solrException
 
-            val exception = shouldThrowExactly<IndexingException> {
-                indexingService.reindex(ZoekObjectType.ZAAK)
+            indexingService.reindex(ZoekObjectType.ZAAK)
+
+            Then("ignores errors") {
+                verify(exactly = 1) {
+                    solrClient.deleteById(any<List<String>>())
+                    solrClient.addBeans(any<Collection<*>>())
+                }
             }
+        }
+    }
+
+    Given("Solr indexing exists and zaak information cannot be obtained") {
+        val queryResponse = mockk<QueryResponse>()
+        val documentList = SolrDocumentList().apply {
+            addAll(
+                listOf(
+                    SolrDocument(mapOf("id" to 1)),
+                    SolrDocument(mapOf("id" to 2))
+                )
+            )
+        }
+        every { queryResponse.results } returns documentList
+        every { queryResponse.nextCursorMark } returns CursorMarkParams.CURSOR_MARK_START
+        every { solrClient.query(any()) } returns queryResponse
+        every { solrClient.deleteById(listOf("1", "2")) } returns UpdateResponse()
+
+        val ioException = IOException("IO exception")
+        every { zrcClientService.listZaken(any<ZaakListParameters>()) } throws ioException
 
-            Then("it re-throws the exception") {
-                exception.cause shouldBe solrException
+        When("reading zaak details throws an error") {
+            indexingService.reindex(ZoekObjectType.ZAAK)
+
+            Then("aborts after max sequential errors") {
+                verify(exactly = MAX_SEQUENTIAL_ERRORS) {
```

Comment:
I think it is better not to use actual code in our tests, including constants. I would just use:

```suggestion
                verify(exactly = 20) {
```

here. Always doing this consistently makes sure that our test code is not influenced by our code under test.